src/dynamicprompts/samplers/combinatorial.py

```diff
@@ -63,19 +65,39 @@ def _get_combinatorial_variant(
     generator: Sampler,
     variant_command: VariantCommand,
 ) -> typing.Iterable[str]:
+    from dynamicprompts.samplers.random import RandomSampler
+
     if len(variant_command.variants) == 0:
         return
 
     seen = set()
 
-    for bound in range(variant_command.min_bound, variant_command.max_bound + 1):
-        for combo in variant_command.get_value_combinations(bound):
-            for prompt_arr in _combo_to_prompt(generator, combo):
-                deduped_arr = _dedupe(prompt_arr)
-                correct_size = len(deduped_arr) == bound
-                if correct_size and deduped_arr not in seen:
-                    seen.add(deduped_arr)
-                    yield variant_command.separator.join(deduped_arr)
+    if variant_command.sampling_method == SamplingMethod.RANDOM:
+        new_sampler = RandomSampler(wildcard_manager=generator._wildcard_manager)
+        yield from new_sampler.generate_prompts(variant_command)
```

Comment:
@akx what do you think about implementing it like this?

Reply:
Showers are definitely where the best ideas come from which is why water-use restrictions can be so bad for productivity :-)

Where would `samplers` get defined? Perhaps in `{Random,Combinatorial}PromptGenerator`?

Then instead of creating a specific sampler, the `PromptGenerator` would simply create the base sampler and pass `samplers` into the constructor? That would make creating a sampler more cumbersome but delegates the strategy to the `PromptGenerator`

Reply:
Yeah, maybe... `GenericSampler` that derives from `Sampler`, I guess, that could create the other samplers?

```
gs = GenericSampler(
   ...,
   default_sampling_method=SamplingMethod.Random,
)
```